core/modules/map.rs

```diff
@@ -777,6 +777,10 @@ impl ModuleMap {
       return Err(v8::Global::new(tc_scope, module.get_exception()));
     }
 
+    if module.get_status() != v8::ModuleStatus::Uninstantiated {
+      return Ok(());
+    }
```

Comment:
Actually this seems fishy - do we now have a failing test without this one?

Reply:
This test fails `tests/testdata/run/088_dynamic_import_already_evaluating.ts`.

I looked into it yesterday (copied from slack):

> I think the issue is that we are calling instantiate_module multiple times if you dynamically import the same module multiple times – that never made sense but they added a check in Module::Instantiate to assert that the module wasn't already in the Instantiating state.

So pretty much yes it's incorrect, but it was always incorrect, this just gets things passing again and avoids the check failure.

Reply:
That makes sense, I think what you're talking about is a "linking phase"; before now, we could just instantiate any module and count on V8 to resolve them in the right order, but I guess that is no longer true. Thanks for the explanation. @devsnek could you add a comment around this line with this info?